packages/synapse-react-client/src/components/SetBasicAccessRequirementFields/SetBasicAccessRequirementFields.tsx

```diff
@@ -0,0 +1,103 @@
+import { Alert } from '@mui/material'
+import {
+  AccessRequirement,
+  ManagedACTAccessRequirement,
+} from '@sage-bionetworks/synapse-types'
+import React, { useImperativeHandle, useState } from 'react'
+import { useUpdateAccessRequirement } from '../../synapse-queries'
+import { AccessRequirementTextInstructions } from './AccessRequirementTextInstructions'
+import { AccessorRequirements } from './AccessorRequirements'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import { AccessRequirementWikiInstructions } from './AccessRequirementWikiInstructions'
+
+export type BasicAccessRequirement = Exclude<
+  AccessRequirement,
+  ManagedACTAccessRequirement
+>
+
+export type SetBasicAccessRequirementFieldsHandle = {
+  /* Allow the parent component to trigger saving the AR, so this may be embedded in an arbitrary modal. */
+  save: () => void
+}
+
+export type SetBasicAccessRequirementFieldsProps = {
+  accessRequirement: BasicAccessRequirement
+  onSaveComplete: (
+    /* null when an error has been returned */
+    updatedAr: BasicAccessRequirement | null,
+  ) => void
+}
+
+export const SetBasicAccessRequirementFields = React.forwardRef(
+  function SetBasicAccessRequirementFields(
+    props: SetBasicAccessRequirementFieldsProps,
+    ref: React.ForwardedRef<SetBasicAccessRequirementFieldsHandle>,
+  ) {
+    const { accessRequirement, onSaveComplete } = props
+    const [updatedAr, setUpdatedAr] =
+      useState<BasicAccessRequirement>(accessRequirement)
+
+    const [isDeletingTextInstructions, setIsDeletingTextInstructions] =
+      useState<boolean>(false)
```

Comment:
`useUpdateAccessRequirement` can be triggered by an action within this component (when the user deletes the legacy text instructions) or from an action in the parent (when the user clicks "Save" in the wizard). Currently, the wizard uses `onSave`/`onError` to decide whether to advance to the next step. I was planning to only call `onSave`/`onError` when the update was triggered by the wizard calling save, since deleting the text instructions shouldn't change the step in the wizard. `isDeletingTextInstructions` distinguishes between these two update types, i.e should the wizard be notified or not. I'm not tied to this behavior though -- is there a better way to handle these cases?

Reply:
Got it. Maybe it would be cleaner to have a handler that accepts `isDeletingTextInstructions` as a parameter rather than changing state, then fire your `onSuccess` behavior in the handler instead of using the `onSuccess` mutation arg. Something like

```ts
// note - switched to use `mutateAsync`
const { mutateAsync: updateAccessRequirement, error: updateArError } =
  useUpdateAccessRequirement<BasicAccessRequirement>({
    onError: () => {
      setIsDeletingTextInstructions(false)
      onError()
    },
  })

async function onUpdateAccessRequirement(updatedAr: AccessRequirement, isDeletingTextInstructions: boolean) {
  await updateAccessRequirement(updatedAr)
  if (isDeletingTextInstructions) {
    setUpdatedAr(updatedAr)
  } else {
    onSave()
  }
}

```